src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -4637,7 +4637,7 @@ trait Types
       && (variance.isContravariant || isSubType(t1, t2, depth))
     )
 
-    corresponds3(tps1, tps2, mapList(tparams)(_.variance))(isSubArg)
+    sameLength(tps1, tps2) && forall3(tps1, tps2, mapList(tparams)(_.variance))(isSubArg)
```

Comment:
Note that `mapList` is not necessary here but first I would like to know the effect of `corresponds3` alone.